app/Decsys/Data/Entities/UserWordlist.cs

```diff
@@ -0,0 +1,14 @@
+using Decsys.Models.Wordlist;
+using MongoDB.Bson;
+
+namespace Decsys.Data.Entities;
+
+public class UserWordlist
+{
+    public ObjectId Id { get; set; }
```

Comment:
Just a question on `ObjectId` - as it's not used for Ids elsewhere in Decsys, probably due to the Mongo/Litedb implementation? Or is it okay to use it like this?

But if this entity is exclusively for Mongo it should then be in that entity folder also.

Reply:
Using int Id started the Id at 0, and then you have to manually update them, where as mongoDb objectId makes more sense as it creates a unique 12 byte identifier.

Reply:
Oooh. First of all great q and second of all great catch.

So to answer the q: the type of id's used in DECSYS is (mostly) fairly carefully thought out, but it's not always clear due to differences in how mongo/litedb behave compared with SQL / EF in particular.

- Any document in a collection even with no apparent `Id` property in the Entity gets automatically given an `ObjectId` `_id` field by Mongo / LiteDb
- Decsys does actually use `ObjectId` (not very noticeably) for the `DecsysUser` Entity, which ultimately inherits from an `IdentityUser` with an `ObjectId` `Id`
- Neither Mongo nor LiteDb autoincrement integer id's by default, and for some Entities (e.g. Pages, Components, and now Wordlists) we want autogenerated unique ids.
    - `ObjectId` would actually be a great fit for this, but at the time I was writing LiteDb code, not Mongo, and I didn't know it had its own `ObjectId` so I used .NET's GUID and abused property initialisers.
    - a side effect of this is that the types using the GUID Id's are still portable between Mongo and LiteDb though...
- There's value in using integers sometimes. Survey and SurveyInstance use integer id's deliberately *despite* the annoyance of Mongo / LiteDb not providing built in autoincrementing, because we want them to be easily encodable into friendly access URLs (where we encode the integer id of each into a base35 string and concatenate them with `z`)

With all that in mind, we decided for Wordlists that ObjectId was a good fit in part as we are really only implementing this for Mongo, and also I had forgotten how Pages / Components were done 😅

The upshot of all this is:

- If I did it all again, or if it were Mongo only, I'd use ObjectId's instead of GUIDs
- I guess now we need to decide whether to consistently use GUID, or use native ObjectId this time.

If we use GUID:
- update the type
- ensure it initialises on entity construction

If we use ObjectId:
- good spot, yes that makes it a Mongo specific entity, so it needs to move

Reply:
> But my vote would be for using ObjectId, given that's the ideal state.

Think I agree.

> If we were to add this functionality to workshop mode (excising Litedb feels more likely though), it looks straightforward to handle it gracefully.

Absolutely, we have an existing pattern for that with other db implementation specific entities today. :)